pkg/controller/directvolumemigration/rsync.go

```diff
@@ -159,16 +159,77 @@ func (t *Task) getRsyncClientMutations(client compat.Client, namespace string) (
 	if err != nil {
 		return nil, liberr.Wrap(err)
 	}
-	runAsUser := int64(0)
-	trueBool := bool(true)
-	customSecurityContext := &corev1.SecurityContext{
-		Privileged:             &isPrivileged,
-		RunAsUser:              &runAsUser,
-		ReadOnlyRootFilesystem: &trueBool,
-		Capabilities: &corev1.Capabilities{
-			Drop: []corev1.Capability{"MKNOD", "SETPCAP"},
-		},
+
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+
+	migration, err := t.Owner.GetMigrationForDVM(destClient)
+	if err != nil {
+		return nil, liberr.Wrap(err)
+	}
+	customSecurityContext := &corev1.SecurityContext{}
+	if migration.Spec.RunAsRoot != nil && *migration.Spec.RunAsRoot {
+		// if rsync needs to run as root
+		checkLabels, err := checkUserNamespaceLabel(client)
+		if err != nil {
+			return nil, liberr.Wrap(err)
+		}
+		privilegedLabelPresent := true
+		if checkLabels {
+			privilegedLabelPresent, err = isPrivilegedLabelPresent(client, namespace)
+			if err != nil {
+				return nil, liberr.Wrap(err)
+			}
+		}
+		if privilegedLabelPresent {
+			trueBool := bool(true)
+			runAsUser := int64(0)
+			customSecurityContext = &corev1.SecurityContext{
+				Privileged:             &isPrivileged,
+				ReadOnlyRootFilesystem: &trueBool,
+				RunAsUser:              &runAsUser,
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"MKNOD", "SETPCAP"},
+				},
+			}
+		} else {
+			runAsNonRoot := bool(true)
+			customSecurityContext.RunAsNonRoot = &runAsNonRoot
+			customSecurityContext.Capabilities = &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			}
+			rsyncMessage := "missing required labels on source namespace, running rsync as non root even though it needs to run as root"
+			t.Log.Info(rsyncMessage)
+			t.Owner.Status.SetCondition(migapi.Condition{
+				Type:     RsyncServerPodsRunningAsNonRoot,
+				Status:   migapi.True,
+				Reason:   "RsyncOperationsAreRunningAsNonRoot",
+				Category: Warn,
+				Message:  rsyncMessage,
+			})
+		}
+	} else if migration.Spec.RunAsGroup != nil || migration.Spec.RunAsUser != nil {
```

Comment:
`else if` here doesn't make sense. User config should _always_ override default values.

Reply:
I am not sure what you mean here. This else if basically is for the case where `RunAsRoot` is not `true` and user has specified `uid/gid` for rsync to run with

Reply:
I think I got it! I updated the code. ty